### ISSUE TITLE
Red and blue color channels switched in hardware rendering mode

### ISSUE THREAD

**Issue by schober-ch** [any1/wlvncc#39]
Hey.

Thank you for this great project! I am trying to access a VNC server running on an Android tablet (https://github.com/bk138/droidVNC-NG) using wlvncc as client on a Linux Laptop with Archlinux and Hyprland. 

It works very well, but when starting `wlvncc xxx.xxx.xxx` with default settings, the red and blue color channels seem swapped (see screenshot 1 below).
This seems to be a known behavior for some combinations of VNC servers and clients (see for example the issue https://github.com/bk138/droidVNC-NG/issues/115 in the vnc server repo). Since unfortunately I'm not an expert on VNC at all, I can not understand if there is some "fault" or if this is more like undefined behavior that different clients (and servers?) have. 

What is interesting (from my non-expert-point-of-view) is that if I start wlvncc with "software rendering" using `wlvncc xxx.xxx... -s`, the error goes away (see 2nd screenshot below).

Now I wonder: Is there something that can be done to fix this behavior in hardware render mode? If you have some idea or hint I'm happy to try some code changes myself, but I would need to get some understanding of all the vnc stuff.. 
 
![image](https://github.com/user-attachments/assets/b769b3fe-e9e3-4054-bf32-d0b4746b0a8b)

![image](https://github.com/user-attachments/assets/287766ad-7de6-4373-b9c0-b4e0ff3f4754)

**Comment by any1** [comment]
Yeah, this is a result of either the server or the client using the wrong pixel format.

E.g. the server might be sending RGB, but the client might think that it's BGR or just render it wrong regardless.

**Comment by any1** [comment]
> And I assume there is no "right" or "wrong" within the VNC protocol, correct?

No, this is well defined and one end is definitely wrong.

> What I'm not sure about is if it makes sense to look into this (since I suspect changing this behavior may break other server/client combinations?) or if it would be an option to create another command line flag for BGR vs RGB? (if this is even possible).

Everything should be implemented to satisfy the protocol, regardless of quirky behaviour in other software. Workarounds for quirks should be explicit. I.e there would have to be a command line option or an environment variable to enable certain workarounds.

Most likely, there's just a wrong or missing conversion somewhere in the EGL code path.

**Comment by schober-ch** [comment]
I see. I probably shouldn't, but I just started to test a little bit (disclaimer: I have no real C-experience and no idea about wayland or vnc). I might look at the completely wrong part, but when using EGL I see a pixel-format(?) value of `875709016` (xBGR) at image.format (https://github.com/any1/wlvncc/blob/1a604623b44eb13f20d2e8d0e7142995499418a1/src/main.c#L423).

But when started using `-s` I get a different value: `875713112`. I could not find which format this is, but the integer is references in a number of wayland and hyprland-related issues.

So I did the obvious and hard-coded the "working" value in the struct
```
	struct image image = {
		...
		.format = 875713112,//w->back_buffer->format,
		...
	};
```
and checked the result - and it is working! 

But now I a) need to stop for today and b) have no clue what to do with that kind of knowledge (other than fixing my problem by hardcoding the value :-D ) . I assume I might dig a little bit more into the code and try to understand where the value is coming from and what its doing, but if you have some short hint or tip what I could do with that I would be grateful :-)
